osu.Game.Rulesets.Mania/Difficulty/Skills/Strain.cs

```diff
@@ -4,98 +4,281 @@
+ * │ Long Note     │      [===]  │      [===]  │      [===]  │       [===] │      [===]   │
+ * ├───────────────┼─────────────┼─────────────┼─────────────┼─────────────┼──────────────┤
+ * │ Long Note (F) │             │             │ (F3)        │ (F4)        │ (F5)         │
+ * │ Before        │      X      │      X      │      [=]    │       [===] │      [=====] │
+ * │ Long Note     │             │             │      [===]  │       [===] │      [===]   │
+ * └───────────────┴─────────────┴─────────────┴─────────────┴─────────────┴──────────────┘
+ */
+
+/* Strain Calculation Intuition
+ *
+ * How Global, Column and Final Strains are calculated
+ *
+ * E.g. Given a map with Columns 0 0 1 2 1 0
+ * 1) Note that the first note is excluded!
+ * 2) f   denotes a strain decay, then strain add operation given the current note.
+ *        f is different between global and column strain.
+ * 3) (+) denotes an add operation.
+ *
+ *                       t=0           t=1           t=2           t=3           t=4           t=5
+ * ┌───────────────────┬─────┐       ┌─────┐       ┌─────┐       ┌─────┐       ┌─────┐       ┌─────┐
+ * │ Global Strain     │  .  ├───f───►  .  ├───f───►  .  ├───f───►  .  ├───f───►  .  ├───f───►  .  │
+ * └───────────────────┴─────┘       └──┬──┘       └──┬──┘       └──┬──┘       └──┬──┘       └──┬──┘
+ *                                      │             │             │             │             │
+ *                                     (+)           (+)           (+)           (+)           (+)
+ *                                      │             │             │             │             │
+ * ┌───────────────────┬─────┐       ┌──▼──┐          │             │             │          ┌──▼──┐
+ * │ Column Strain 0   │  .  ├───f───►  .  ├──────────┼─────────────┼─────────────┼──────f───►  .  ├───►
+ * ├───────────────────┴─────┤       └──┬──┘          │             │             │          └──┬──┘
+ * ├───────────────────┬─────┤          │          ┌──▼──┐          │          ┌──▼──┐          │
+ * │ Column Strain 1   │  .  ├──────────┼──────f───►  .  ├──────────┼──────f───►  .  ├──────────┼──────►
+ * ├───────────────────┴─────┤          │          └──┬──┘          │          └──┬──┘          │
+ * ├───────────────────┬─────┤          │             │          ┌──▼──┐          │             │
+ * │ Column Strain 2   │  .  ├──────────┼─────────────┼──────f───►  .  ├──────────┼─────────────┼──────►
+ * └───────────────────┴─────┘          │             │          └──┬──┘          │             │
+ *                                      │             │             │             │             │
+ * ┌─────────────────────────┐       ┌──▼──┐       ┌──▼──┐       ┌──▼──┐       ┌──▼──┐       ┌──▼──┐
+ * │ Final Strain            │       │  .  │       │  .  │       │  .  │       │  .  │       │  .  │
+ * └─────────────────────────┘       └─────┘       └─────┘       └─────┘       └─────┘       └─────┘
```

Comment:
I would not use unicode characters here. It obliterates the formatting you are going for here on some fonts.

![1687108229](https://github.com/ppy/osu/assets/20418176/9bc0d47d-dc61-4959-83b2-630bc097373a)

At this point I'm not even sure this should live in source. Maybe this should be a markdown document next to the source or something.

Reply:
sure, I'll make this a `README.md` on the same dir. 
In retrospect, it's a hassle to scroll up and down the code